File: app/api/v1/navigation.py
from fastapi import APIRouter, Depends, HTTPException, Query

from app.api.deps import get_current_active_user
from app.models.user import User
from app.services.navigation_service import fetch_route

router = APIRouter(prefix="/navigation", tags=["navigation"])

@router.get("/route")
def get_route(
    origin_lat: float = Query(...),
    origin_lng: float = Query(...),
    dest_lat: float = Query(...),
    dest_lng: float = Query(...),
    current_user: User = Depends(get_current_active_user),
):
    try:
        return fetch_route(origin_lat, origin_lng, dest_lat, dest_lng)
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"route api error: {e}")

File: src/api/navigation.js
import client from './client';

export async function getRoute(originLat, originLng, destLat, destLng) {
  const response = await client.get('/navigation/route', {
    params: {
      origin_lat: originLat,
      origin_lng: originLng,
      dest_lat: destLat,
      dest_lng: destLng,
    },
  });
  return response.data;
}
